phc/util/csv_writer.py

```diff
@@ -44,10 +49,7 @@ def write(self, frame: pd.DataFrame):
         self._finalize()
 
     def _columns(self):
-        with open(self.filename, "r") as f:
-            header = f.readline()
-
-        return pd.read_csv(StringIO(header)).columns.tolist()
+        return pd.read_csv(self.filename, nrows=0).columns.tolist()
```

Comment:
This will actually be a major problem. We keep the memory (roughly) constant by not reading the whole file. Whatever our solution is, I don't think we can have this as an option. 😬

Reply:
That call is specifying `nrows=0`. Does pandas actually read the entire file even when this option is specified?

Reply:
Ah, I see now. That looks fine then according to the docs.